debezium-core/src/main/java/io/debezium/relational/Key.java

```diff
@@ -0,0 +1,154 @@
+    
+    
+    private Key(Table table, KeyMapper keyMapper) {
+        this.table = table;
+        this.keyMapper = keyMapper;
+    }
+
+    /**
+     * 
+     * @return the columns {@link Column} taking part of the key.
+     */
+    public List<Column> keyColumns() {
+        return keyMapper.matches(table);
+    }
+    
+    public static class Builder {
+        
+        private final Table table;
+        private KeyMapper keyMapper = IdentityKeyMapper.getInstance();
+        
+        public Builder( Table table) {
+            this.table = table;
+        }
+        
+        public Builder customKeyMapper(CustomKeyMapper customKeyMapper) {
+            if (customKeyMapper != null) {
+                this.keyMapper = customKeyMapper;
+            }
+            return this;
+        }
+        
+        public Key build() {
+            return new Key(table, keyMapper);
+        }
+        
+    }
+    
+    @FunctionalInterface
+    public static interface KeyMapper {
+        
+        /**
+         * 
+         * @param table {@code Table} 
+         * @return the list {@code Column} describing the key.
+         */
+        List<Column> matches(Table table);
+        
+    }
+    
+    /**
+     * 
+     * Default Key mapper using PK as key.
+     *
+     */
+    @FunctionalInterface
+    private static interface IdentityKeyMapper extends KeyMapper {
+        
+        public static IdentityKeyMapper getInstance() {
+            return (table) -> {
+                return table.columns()
+                        .stream()
+                        .filter(c -> table.isPrimaryKeyColumn(c.name()))
+                        .collect(Collectors.toList());
+            };
+        }
+        
+    }
+    
+    /**
+     * 
+     * Custom Key mapper used to override or defining a custom {@code Key}
+     *
+     */
+    @FunctionalInterface
+    public static interface CustomKeyMapper extends KeyMapper {
+        
+        /**
+         * Getting an instance with a list of regexp (table:column1,column2) delimited by ';' matching the tables keys. 
+         * ex: inventory.customers:pk1,pk2;(.*).purchaseorders:pk3,pk4
+         * 
+         * @param fullyQualifiedColumnNames a list of regex 
+         * @return a new {@code CustomKeyMapper} or null if fullyQualifiedColumnNames is invalid.
+         */
+        public static CustomKeyMapper getInstance(String fullyQualifiedColumnNames) {
+            
+            if (fullyQualifiedColumnNames == null) {
+                return null;
+            }
+            
+            // transform the 'message.key.columns' option into a list of regexp. 
+            // ex: message.key.columns=inventory.customers:pk1,pk2;(.*).purchaseorders:pk3,pk4
+            // will become => [inventory.customers.pk1,inventory.customers.pk2,(.*).purchaseorders.pk3,(.*).purchaseorders.pk4]
+            // then joining those values 
+            String regexes =  Arrays.asList(fullyQualifiedColumnNames.split(";"))
```

Comment:
Note this would break if one of the patterns itself would contain a ";", e.g. within a table name. I can live with it though, it seems unlikely enough.

Reply:
I agree that it is not perfect and prone to some errors. Even the regex validation does not seem to match all the cases (I should probably improve it but I am really not a regex expert).